src/plugins/intel_cpu/src/infer_request.cpp

```diff
@@ -95,48 +96,26 @@ void InferRequestBase::pushInput(const std::string& inputName, InferenceEngine::
     graph->PushInputData(inputName, needConvert ? iconv : inputBlob);
 }
 
-void InferRequestBase::PushStates() {
+// state -> node
+void InferRequestBase::AssignStates() {
     for (auto &node : graph->GetNodes()) {
         if (node->getType() == Type::MemoryInput) {
-            auto cur_node = dynamic_cast<node::MemoryInput*>(node.get());
+            auto cur_node = std::dynamic_pointer_cast<node::MemoryInput>(node);
```

Comment:
Could be avoided by saving the `MemInput` nodes in the graph in a separate container, just like for the I/O nodes.